### PR TITLE
nixos/murmur: log to systemd journal, expose wrapper

### PR DESCRIPTION
**Pull request by florianjacob**
###### Motivation for this change
Everything I have on my server logs to systemd journal by default - except murmur.
This takes murmur in line with all other services by changing that.

On the very first start, murmur writes the password of the SuperUser into the log file.
To change this if desired or if that entry is not available anymore, this exposes a wrapper combining the binary and the service config file as the `murmurd-service` executable. (open to better naming propositions!).

As long as it's in the journal, the initial SuperUser password is now readable for all users in the `journal` group instead of `root` and `murmur` only. In case that group is not equal with the murmur administrators, the password can be changed with the new wrapper.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by florianjacob** [comment]
@GrahamcOfBorg test mumble

**Comment by florianjacob** [comment]
Seems like I don't have the required permissions to run the tests through ofBorg, and when I try to run the tests locally I get a kernel panic, which does not seem to be related to the tests themselves, though.

**Comment by arianvp** [comment]
@GrahamcOfBorg test mumble

File: nixos/modules/services/networking/murmur.nix [hunk 1]
```diff
@@ -251,19 +273,13 @@ in
       after       = [ "network-online.target "];
 
       serviceConfig = {
-        Type      = "forking";
+        Type      = "simple";
```
Comment (aanderse) [inline nixos/modules/services/networking/murmur.nix]: Simple is the default type so this isn't actually necessary.

File: nixos/modules/services/networking/murmur.nix [hunk 2]
```diff
@@ -251,19 +273,13 @@ in
...
         RuntimeDirectory = "murmur";
-        PIDFile   = cfg.pidfile;
         Restart   = "always";
         User      = "murmur";
-        ExecStart = "${pkgs.murmur}/bin/murmurd -ini ${configFile}";
+        ExecStart = "${cli-wrapper}/bin/murmurd-service -fg";
         PermissionsStartOnly = true;
```
Comment (aanderse) [inline nixos/modules/services/networking/murmur.nix]: I know https://github.com/NixOS/nixpkgs/pull/56265 isn't related to this PR at all... but would you be willing to do me a solid and remove `PermissionsStartOnly = true;` in an extra commit here? It looks like the only other things to do would be to add `StateDirectory = "murmur";` and remove the call to `createHome = true;` and you would be good to go...

**Review by aanderse — COMMENTED** [review]
Feel free to completely ignore my request to remove `PermissionsStartOnly = true;`... but it would be super awesome if you added an extra commit doing so! :smile:

**Comment by xaverdh** [comment]
Since https://github.com/NixOS/nixpkgs/pull/38775 was merged, I guess this can be closed now?

**Pull request closed without merging**